client/my-sites/stats/stats-period-navigation/index.jsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'components/gridicon';
+
+const StatsPeriodNavigation = ( { url, children, date, period, moment } ) => {
+	const isToday = moment( date ).isSame( moment(), period );
+	const previousDay = moment( date ).subtract( 1, period ).format( 'YYYY-MM-DD' );
```

Comment:
Yeah! I thought about this, and my conclusion for now, is that it may add some complexity to the component. We'll need to fetch the stats on the navigation component to know when this period starts. 

It should be easier to achieve when we reduxify all these endpoints.